### PR TITLE
drivers: flash: stm32h7: fix clearing of erased sector number

### PR DESCRIPTION
**Pull request by ghost**
This fixes the logic in erase_sector() for clearing the previous
sector number. The logic was reversed, clearing bits it should not
clear and keeping previous bits of the sector number. In practice this
does not seem to have had any effect on the current functionality, but
it will start to matter later if e.g. flash interrupts are enabled.

### CODE REVIEW

**Comment by erwango** [comment]
@nicovince, can you have a look ?

File: drivers/flash/flash_stm32h7x.c [hunk 1]
```diff
@@ -235,7 +235,7 @@ static int erase_sector(const struct device *dev, int offset)
 		return rc;
 	}
 
-	*(sector.cr) &= FLASH_CR_SNB;
+	*(sector.cr) &= ~FLASH_CR_SNB;
```
Comment (erwango) [inline drivers/flash/flash_stm32h7x.c:238]: Could be nice adding a comment, just to remember this was done on purpose.

**Comment by nvincent-vossloh** [comment]
Ok for me.

**Review by erwango — APPROVED** [review]

**Comment by erwango** [comment]
@nvincent-vossloh Could you approve formally ?

**Review by nvincent-vossloh — APPROVED** [review]

**Comment by nvincent-vossloh** [comment]
> @nvincent-vossloh Could you approve formally ?

Just did, but apparently it requires an authorized reviewer :)

**Comment by erwango** [comment]
> > @nvincent-vossloh Could you approve formally ?
> 
> Just did, but apparently it requires an authorized reviewer :)

Thanks, that will do.

**Review by nvlsianpu — APPROVED** [review]

**Pull request merged**